File: Array/06_left_roste_array_by_d_place/array_by_d_place.js

// brute aproch
function leftRotate(arr, d) {
  let n = arr.length;
  d = d % n;
  let temp = new Array(d);
  for (let i = 0; i < d; i++) {
    temp[i] = arr[i];
  }
  for (let i = d; i < n; i++) {
    arr[i - d] = arr[i];
  }
  for (let i = n - d; i < n; i++) {
    arr[i] = temp[i - (n - d)];
  }
  return arr;
}

arr = [1, 2, 3, 4, 5, 6, 7];

console.log(leftRotate(arr, 3));

 

File: Array/06_left_roste_array_by_d_place/array_by_d_place.py
def roatae_array(arr ,d):
    n = len(arr)
    d = d%n
    temp = [0]*d
    for i in range(0,d):
        temp[i] = arr[i]
    for i in range(d,n):
        arr[i-d] = arr[i]
    for  i in range( n-d , n):
        arr[i] = temp[i-(n-d)]
    

arr = [1,2,3,4,5,6,7]

print(roatae_array(arr,3))



